miraclelinux/meta-debian
# tzdata: Update version from 2021a-0+deb10u11 to 2021a-0+deb10u12

# Purpose of pull request
This version updated timezone database for leap second list.

DLA: DLA 3684-1

# Test
1. Build package
2. Run image and test the package on the qemuarm64 environment

## How to test the package
1. Build qemuarm64 image
2. Check leap seconds list changes on source tree
3. Run qemu image
4. Check the files installed by the tzdata package

### Build qemuarm64 image
Add the following to local.conf and build qemuarm64 image.
```
IMAGE_INSTALL_append = " tzdata"
```
```
$ bitbake core-image-minimal
```

### Check leap seconds list changes on source tree
```
$ grep 'Update' ./tmp/work/all-deby-linux/tzdata/2021a-r0/leap-seconds.list 
```

### Run qemu image
```
$ runqemu qemuarm64 nographic
```

### Check the files installed by the tzdata package
```
# cd /usr/share/zoneinfo/
# ls -F
```

## Test result
### Build package
Build succeeded.
```
$ bitbake tzdata
Parsing recipes: 100% |##############################################################################| Time: 0:01:19
Parsing of 1041 .bb files complete (0 cached, 1041 parsed). 1821 targets, 73 skipped, 0 masked, 0 errors.
NOTE: Resolving any missing task queue dependencies

Build Configuration:
BB_VERSION           = "1.42.0"
BUILD_SYS            = "x86_64-linux"
NATIVELSBSTRING      = "debian-10"
TARGET_SYS           = "aarch64-deby-linux"
MACHINE              = "qemuarm64"
DISTRO               = "deby"
DISTRO_VERSION       = "10.0"
TUNE_FEATURES        = "aarch64 armv8a crc"
TARGET_FPU           = ""
meta                 
meta-poky            = "warrior:d4b57c68b22027c2bedff335dee06af963e4f8a8"
meta-debian          = "update-tzdata_2021a-0+deb10u12:bd27f9d8b98de669e2d4901ebd926ce82cf427af"

NOTE: Fetching uninative binary shim from http://downloads.yoctoproject.org/releases/uninative/2.9/x86_64-nativesdk-libc.tar.xz;sha256sum=d07916b95c419c81541a19c8ef0ed8cbd78ae18437ff28a4c8a60ef40518e423
Initialising tasks: 100% |###########################################################################| Time: 0:00:08
Sstate summary: Wanted 52 Found 0 Missed 52 Current 0 (0% match, 0% complete)
NOTE: Executing SetScene Tasks
NOTE: Executing RunQueue Tasks
NOTE: Tasks Summary: Attempted 462 tasks of which 0 didn't need to be rerun and all succeeded.
```

### Package test
The leap seconds list has been updated.
```
$ grep 'Update' ./tmp/work/all-deby-linux/tzdata/2021a-r0/leap-seconds.list 
#	Last Update of leap second values:   8 July 2016
#	Updated through IERS Bulletin C66
```

The tzdata package are installed.
```
# cd /usr/share/zoneinfo/
# ls -F
Africa/       EST           HST           NZ            US/
America/      EST5EDT       Hongkong      NZ-CHAT       UTC
Antarctica/   Egypt         Iceland       Navajo        Universal
Arctic/       Eire          Indian/       PRC           W-SU
Asia/         Etc/          Iran          PST8PDT       WET
Atlantic/     Europe/       Israel        Pacific/      Zulu
Australia/    Factory       Jamaica       Poland        iso3166.tab
Brazil/       GB            Japan         Portugal      posix/
CET           GB-Eire       Kwajalein     ROC           right/
CST6CDT       GMT           Libya         ROK           zone.tab
Canada/       GMT+0         MET           Singapore     zone1970.tab
Chile/        GMT-0         MST           SystemV/
Cuba          GMT0          MST7MDT       Turkey
EET           Greenwich     Mexico/       UCT
```
The tzdata package are installed.
